grid-proxy/internal/indexer/health.go

```diff
@@ -0,0 +1,139 @@
+package indexer
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-proxy/internal/explorer/db"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-proxy/pkg/types"
+	"github.com/threefoldtech/tfgrid-sdk-go/rmb-sdk-go/peer"
+)
+
+const (
+	healthCallCmd = "zos.system.version"
+)
+
+type NodeHealthIndexer struct {
+	database        db.Database
+	rmbClient       *peer.RpcClient
+	nodeTwinIdsChan chan uint32
+	resultChan      chan types.HealthReport
+	batchChan       chan []types.HealthReport
+	indexerInterval time.Duration
+	indexerWorkers  uint
+	batchSize       uint
+}
+
+func NewNodeHealthIndexer(
```

Comment:
I can see that the an indexer mainly does the same operation, I was wondering that an indexer logic can be abstracted more. 
From what i see an indexer has a set of options (say workers), he run indexing every certain amount of time. Then does some sort of batching to update the db. 
All this logic then can be unified. Then each indexer need to define 2 things.
- The function that does the "fetch" of the data. 
- The logic of "batch" db update.

Those 2 functions then can be part of an IndexerWork interface. And can has multiple implementation (say gpu and dmi) and then an Indexer instance can be created for each something like `NewIndexer(GPUIndexerWork())` for example. Then call `indexer.Start()`